packages/knip/src/plugins/relay/index.ts

```diff
@@ -0,0 +1,60 @@
+import parseArgs from 'minimist';
+import type { IsPluginEnabled, Plugin, ResolveEntryPaths } from '../../types/config.js';
+import { toDeferResolve, toEntry } from '../../util/input.js';
+import { isInternal, join } from '../../util/path.js';
+import { hasDependency } from '../../util/plugin.js';
+import type { RelayConfig } from './types.js';
+// https://relay.dev/docs/next/guides/compiler/#configuration
+// https://github.com/facebook/relay/blob/main/compiler/crates/relay-compiler/relay-compiler-config-schema.json
+
+const title = 'Relay';
+
+const enablers = ['vite-plugin-relay', '@swc/plugin-relay', 'babel-plugin-relay'];
```

Comment:
```suggestion
const enablers = ['react-relay', 'vite-plugin-relay', '@swc/plugin-relay', 'babel-plugin-relay'];
```

> Some people use artifacts with fetch client other than `react-relay` so this is not a good enabler.

Why is `react-relay` not a good enabler? If that dependency is listed, the plugin should be enabled, right? In what situation is that not the case?

Reply:
1. Adding all `*.graphql.ts` files would cause issues if user named his own files `*.graphql.ts`
2. The documentation u are linking leads to `rescript` Lang version of relay, their compiler is adjusted to work with `rescript`. In either case the object might not have a `path` prop, it can be a simple string representing a typescript type `number` | `Record<...,  ...>`. I accounted for that, they are filtered out. In the case of rescript they use a string to access module `ModuleName.type`, this shouldn't cause any issues.
3. Ignoring custom scalars all together is an interesting idea. I had them added to knip ignore in my project and just assumed plugin should handle this for me.

Not adding them as entrypoints would require user to explicitly ignore these which can be useful information.

For me this would just be more annoying and increase iteration time, for example a PR will remove a field - now the type is unused, knip will fail in ci, file has to be added to ignore. I'd add all the scalars to ignore/entrypoint manually anyways, and then there's a chance these will get out of sync. The reason I enabled `requireCustomScalarTypes` is so I have these types ready and I can select/remove fields and don't worry about anything missing.

Reply:
1. Sorry I wasn't clear, what I meant was what we're already doing: `toEntry(join(artifactDirectory, '**')` - it was more of a prerequisite prior to the actual proposal to ignore the scalar paths.
2. Regardless of Rescript, if the `path` prop isn't there then `toDeferResolve(customScalarType.path)` means `toDeferResolve(undefined)` - which might eventually not be an issue, but isn't great either?
3. This might not fit well in your workflow, but reporting unused files etc. is what Knip does. Compare this to dependencies in `package.json` or named exports of a file. The idea is that the unused ones are reported (not to keep them "just in case"). Is there maybe a way this could fit in your workflow better e.g. by naming conventions that you can then `ignore` or something?

Reply:
I like what you are proposing, even with the option enabled the user will be warned about unused files and can make their decision to ignore them or delete. 

I already have my files ignored so it's not a big deal for me.

Reply:
done this as well as other suggestions